File: app/src/main/java/me/rosuh/jianews/view/IView.java
package me.rosuh.jianews.view;

import java.util.List;

import me.rosuh.jianews.bean.ArticleBean;

/**
 * @author rosu
 * @date 2018/9/30
 */
public interface IView {
    void onHeaderRequestFinished(List<ArticleBean> list);

    void onUpdateDataFinished(List<ArticleBean> list, int nextPos);

    void onUpdateDataFailed(Throwable t);

    void scrollToTop();
}

File: app/src/main/java/me/rosuh/jianews/view/IView.kt
package me.rosuh.jianews.view

import me.rosuh.jianews.bean.ArticleBean

/**
 * @author rosu
 * @date 2018/9/30
 */
interface IView {

    fun onHeaderRequestFinished(list: ArrayList<ArticleBean>)

    fun onUpdateDataFinished(list: ArrayList<ArticleBean>, nextPos: Int)

    fun onUpdateDataFailed(t: Throwable)

    fun scrollToTop()
}
